jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ReHandlingErrorProcessor.java

```diff
@@ -0,0 +1,146 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.server.handler;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.eclipse.jetty.http.HttpURI;
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.ExceptionUtil;
+import org.eclipse.jetty.util.URIUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An ErrorProcessor that can re-handle a request at an error page location.
+ */
+public abstract class ReHandlingErrorProcessor extends ErrorProcessor
+{
+    private static final Logger LOG = LoggerFactory.getLogger(ReHandlingErrorProcessor.class);
+
+    private final Handler _handler;
+
+    protected ReHandlingErrorProcessor(Handler handler)
+    {
+        _handler = handler;
+    }
+
+    @Override
+    public InvocationType getInvocationType()
+    {
+        return _handler.getInvocationType();
+    }
+
+    @Override
+    protected void generateResponse(Request request, Response response, int code, String message, Throwable cause, Callback callback) throws IOException
+    {
+        if (request.getAttribute(ReHandlingErrorProcessor.class.getName()) == null)
+        {
+            String pathInContext = getReHandlePathInContext(request, code, cause);
+
+            if (pathInContext != null)
+            {
+                request.setAttribute(ReHandlingErrorProcessor.class.getName(), pathInContext);
+                HttpURI uri = HttpURI.build(request.getHttpURI()).path(URIUtil.addPaths(request.getContext().getContextPath(), pathInContext)).asImmutable();
+                Request.Wrapper wrapper = new ReHandleRequestWrapper(request, uri, uri.getCanonicalPath());
+
+                try
+                {
+                    Request.Processor processor = _handler.handle(wrapper);
+                    if (processor != null)
+                    {
+                        response.setStatus(200);
```

Comment:
Why setting the status? I think the processor we are invoking on the next line will do whatever it wants, so no point setting the status here.

Reply:
Because we are dispatching to a normal page, perhaps even just a fixed html page, so we need an initial 200 status like any other dispatch.   This is NOT an error processor at this point, which would be generating an error response with an 4XX or 5XX code.  This is just serving a normal page that might say "you had a 4XX or 5XX error".